### PR TITLE
Create a Godmode-like tool for testing emails #5601

### LINKED ISSUE #5601: Create a Godmode-like tool for testing emails
Currently we test email contents with `expr-1{*}expr-2{*}expr-3{*}...` where `expr-n` are expressions found in the email and `{*}` are wildcards like how we've always been using it. We can test the email content in its entirety with a Godmode-like tool. We can even extend it to the email's metadata (sender, receiver, subject, ... you name it) with relative ease.

To be done after ~~#5593~~ #5585.

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #5601 

Here is the earliest version. Many aspects are open for discussion. Most of the workflow is adapted from the Godmode we have for HTML comparison.
- The trigger keyword is the same, "godmode", which may or may not cause problems.
- The expected page is not HTML-standardized because it is not prone to browser quirks etc. but can be done if there are good reasons for it.
- The email metadata is not included in the HTML page for now (still tested separately as per before), again can be done if there are good reasons.

Once we have Godmode for emails established, I can work on maximizing the test coverage for email generator functions.

### CODE REVIEW

File: src/test/java/teammates/test/cases/logic/EmailGeneratorTest.java [hunk 1]
```diff
@@ -195,23 +132,21 @@ public void testSystemCrashReportEmailContent() {
...
-                             + "</code></pre>"
-                             + "<br><b>Actual user</b>" + "Not logged in"
-                             + "<br><b>Request Method</b>" + requestMethod
-                             + "<br><b>User Agent</b>" + requestUserAgent
-                             + "<br><b>Request Url</b>" + requestUrl
-                             + "<br><b>Request Path</b>" + requestPath
-                             + "<br><b>Request Parameters</b>" + requestParam
-                             + "<br><b>Stack Trace</b><pre><code>" + stackTrace + "</code></pre>";
+        // The stack trace is different depending on the environment in which the test is run at.
+        // As a workaround, after the last common line, change all the stack trace to ...
```
Comment (kanghj) [inline src/test/java/teammates/test/cases/logic/EmailGeneratorTest.java]: Let's wrap `...` in double quotes, `"..."`

File: src/test/java/teammates/test/cases/logic/EmailGeneratorTest.java [hunk 2]
```diff
@@ -35,14 +33,20 @@
     private static final InstructorsLogic instructorsLogic = InstructorsLogic.inst();
     private static final StudentsLogic studentsLogic = StudentsLogic.inst();
     
+    /** indicates if the test-run is to use GodMode */
+    private static boolean enableGodMode;
```
Comment (kanghj) [inline src/test/java/teammates/test/cases/logic/EmailGeneratorTest.java]: Rename to `isEnableGodMode`?

**Comment by kanghj** [comment]
It should be fine to use the same system property `godmode` since the intent and use case is similar. But to prevent future confusion, should we update that `godmode.md` file to mention email tests?

File: src/test/java/teammates/test/cases/logic/EmailGeneratorTest.java (hunk 2 above)
Comment (wkurniawan07) [inline src/test/java/teammates/test/cases/logic/EmailGeneratorTest.java]: Was following the example in `BaseUiTestCase` :p
Guess I get to rename both.

**Pull request merged**